cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx

```diff
@@ -0,0 +1,142 @@
+// React imports
+import React, { useContext, useState } from 'react';
+
+// Library imports
+import { CdsButton } from '@cds/react/button';
+import { CdsIcon } from '@cds/react/icon';
+import { yupResolver } from '@hookform/resolvers/yup/dist/yup';
+import * as yup from 'yup';
+
+// App imports
+
+import { AZURE_FIELDS } from '../AzureManagementCluster.constants';
+import { AzureStore } from '../../../../state-management/stores/Azure.store';
+import { ClusterName, clusterNameValidation } from '../../../../shared/components/FormInputComponents/ClusterName/ClusterName';
+import { INPUT_CHANGE } from '../../../../state-management/actions/Form.actions';
+import {
+    NodeInstanceType,
+    nodeInstanceTypeValidation,
+    NodeProfile,
+} from '../../../../shared/components/FormInputComponents/NodeProfile/NodeProfile';
+import { StepProps } from '../../../../shared/components/wizard/Wizard';
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+// NOTE: icons must be imported
+const nodeInstanceTypes: NodeInstanceType[] = [
+    {
+        id: 'single-node',
+        label: 'Single node',
+        icon: 'block',
+        description: 'Create a single control plane node with a Standard_D2s_v3 instance type',
+    },
+    {
+        id: 'high-availability',
+        label: 'High availability',
+        icon: 'blocks-group',
+        description: 'Create a multi-node control plane with a Standard_D2s_v3 instance type',
+    },
+    {
+        id: 'compute-optimized',
+        label: 'Production-ready (High availability)',
+        icon: 'blocks-group',
+        isSolidIcon: true,
+        description: 'Create a multi-node control plane with a Standard_D4s_v3 instance type',
+    },
+];
+
+type AZURE_CLUSTER_SETTING_STEP_FIELDS = AZURE_FIELDS.CLUSTER_NAME | AZURE_FIELDS.INSTANCE_TYPE;
+
+interface AzureClusterSettingFormInputs {
+    [AZURE_FIELDS.CLUSTER_NAME]: string;
+    [AZURE_FIELDS.INSTANCE_TYPE]: string;
+}
+
+export function AzureClusterSettingsStep(props: Partial<StepProps>) {
+    const { currentStep, deploy, handleValueChange } = props;
+    const { azureState, azureDispatch } = useContext(AzureStore);
+    const azureClusterSettingsFormSchema = yup.object(createYupSchemaObject()).required();
+    const methods = useForm<AzureClusterSettingFormInputs>({
+        resolver: yupResolver(azureClusterSettingsFormSchema),
+    });
+
+    const {
+        handleSubmit,
+        formState: { errors },
+        register,
+        setValue,
+    } = methods;
+
+    let initialSelectedInstanceTypeId = azureState[AZURE_FIELDS.INSTANCE_TYPE];
+    if (!initialSelectedInstanceTypeId) {
+        initialSelectedInstanceTypeId = nodeInstanceTypes[0].id;
+        setValue(AZURE_FIELDS.INSTANCE_TYPE, initialSelectedInstanceTypeId);
+    }
+    const [selectedInstanceTypeId, setSelectedInstanceTypeId] = useState(initialSelectedInstanceTypeId);
+
+    const canContinue = (): boolean => {
+        return Object.keys(errors).length === 0;
```

Comment:
Were you going to enhance this so that you check that cluster name (and instance type) have values?

Reply:
Yes, thanks for the reminder!